dataformat-xml-dom/src/main/java/org/camunda/spin/impl/xml/dom/format/DomXmlDataFormatWriter.java

```diff
@@ -55,28 +60,71 @@ protected void writeResult(StreamResult streamResult, Object input) {
     Node node = (Node) input;
     DOMSource domSource = new DOMSource(node);
     try {
-      getTransformer().transform(domSource, streamResult);
+      if (domXmlDataFormat.isPrettyPrint()) {
+        getFormattingTransformer().transform(domSource, streamResult);
+      } else {
+        getTransformer().transform(domSource, streamResult);
+      }
     } catch (TransformerException e) {
       throw LOG.unableToTransformElement(node, e);
     }
   }
 
+  protected Templates formattingTemplates = null;
+
+  /**
+   * Return a {@link Templates} instance for the strip-spaces.xsl stylesheet.
+   * The Templates instance is initialized lazily in order to use the configured
+   * {@link TransformerFactory} from the {@link DomXmlDataFormat}.
+   *
+   * @return the templates instance for strip-spaces.xsl.
+   */
+  protected synchronized Templates getFormattingTemplates() {
+    if (null == formattingTemplates) {
+      TransformerFactory transformerFactory = domXmlDataFormat.getTransformerFactory();
+      try {
+        try (InputStream xslIn = Thread.currentThread().getContextClassLoader().getResourceAsStream(STRIP_SPACE_XSL)) {
+          Source xslt = new StreamSource(xslIn);
+          formattingTemplates = transformerFactory.newTemplates(xslt);
+        }
+      } catch (TransformerConfigurationException | IOException e) {
+        LOG.unableToCreateTransformer(e);
```

Comment:
If this branch is reached, we will get a `NullPointerException` in `#getFormattingTransformer`. Can we have this method declare `throws ...` and then have exception handling only in `#getFormattingTransformer`?

Reply:
getFormattingTemplates now throws TransformerConfigurationException so handling is moved to getFormattingTransformer as requested